component/identity/src/main/java/org/exoplatform/services/organization/idm/MembershipDAOImpl.java

```diff
@@ -714,6 +715,53 @@ public Membership findMembership(String id) throws Exception {
         return null;
     }
 
+    @Override
+    public List<MembershipType> findMembershipTypesByGroup (String groupId) throws Exception{
+        if (log.isTraceEnabled()) {
+            Tools.logMethodIn(log, LogLevel.TRACE, "findMembershipTypesByGroup", new Object[] { "group", groupId});
+        }
+
+        String plGroupName = getPLIDMGroupName(getGroupNameFromId(groupId));
+
+        String groupKey = getIdentitySession().getPersistenceManager().createGroupKey(plGroupName, getGroupTypeFromId(groupId));
+
+        org.picketlink.idm.api.Group gtnGroup = service_.getIdentitySession().getPersistenceManager().findGroupByKey(groupKey);
+
+        if (gtnGroup == null) {
+            log.log(LogLevel.ERROR, "Internal ERROR. Cannot obtain group: " + groupId);
+            return new ArrayList<>();
+        }
+
+        orgService.flush();
+
+        Collection<RoleType> roleTypes = new HashSet();
+
+        try {
+            roleTypes = getIdentitySession().getRoleManager().findGroupRoleTypes(gtnGroup);
+        } catch (Exception e) {
+            handleException("Identity operation error: ", e);
+        }
+
+        if (log.isTraceEnabled()) {
+            Tools.logMethodOut(log, LogLevel.TRACE, "findMembershipTypesByGroup", roleTypes);
+        }
+
+        List<MembershipType> membershipTypes = new ArrayList<>();
+        if (roleTypes != null && !roleTypes.isEmpty()) {
```

Comment:
IMO CollectionUtils.isEmpty can be used instead

Reply:
The same code will be used for both cases, I prefer avoiding that to depend less on utilities classes as soon as they are not required.